### ISSUE TITLE
Self-Supervised GANs with Label Augmentation

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#143]
## TL; DR

- Devise multi-class classification formulation for discriminator to incorporate SSL to generative adversarial learning.

## Motivation

- Conditional Generation requires class labels; would be nice to model complex joint data distribution without labels.
- GAN Discriminator training resembles online learning setting; distribution of generated samples constantly changes, leading to non-stationarity and *catastrophic forgetting* of the discriminator.
- Adding SSL objective improves stationarity, but potentially introduces goal inconsistencies. (Generator simply produces SSL-friendly samples)

## Prior Works

- SSGAN (Self-Supervised GAN, CVPR'19): Introduce auxiliary rotation loss for the discriminator to mitigate catastrophic forgetting.
- SSGAN-MS: Improved SS-GAN baseline with multi-class minimax (D: recognize rotation angle for real samples and classify the generated sample as fake)

## Method

![image](https://user-images.githubusercontent.com/46308407/147865855-e58c87f0-2ba5-442c-a3fa-a7efd43aba48.png)

![image](https://user-images.githubusercontent.com/46308407/147865862-906d7383-cd9f-482a-ad21-b3b8f28d86df.png)

- Say, there are *k* rotation angles and {real, fake}.
- Discriminator solves *2k* multi-class classification task, recognizing the rotation angle ***and*** real/fake simultaneously.

![image](https://user-images.githubusercontent.com/46308407/147865884-836a060b-94c3-4c83-a078-95cd3dd647f7.png)

- Discriminator objective

![image](https://user-images.githubusercontent.com/46308407/147865889-932df771-9548-46cd-bee5-bae78e4c2c5b.png)

- Generator objective

![image](https://user-images.githubusercontent.com/46308407/147865891-c189fd25-879f-41e1-87ca-924cb97730cc.png)

- Generator objective under optimal discriminator. Hence, under SSGAN-LA (Label Augmented), the generator is encouraged to produce samples indistinguishable from real data points under ***all*** augmentation conditions.
- At the same time, the discriminator is enforced to learn a more stable / semantic representation through auxiliary SSL task.
- Unlike SSGAN and SSGAN-MS where rotation prediction is done only for real samples, the discriminator recognizes augmentation for generated samples as well, jointly considering discrepancy between real and generated data distribution.
- *(personal opinion)* Discriminator is strengthened by expanded support; it looks not only at the real samples, but also fake samples, which are continually mimicking real data. It could be regarded as an additional augmentation, which proves effective especially for data-limited generation regime.

### Baseline

- SSGAN
- SSGAN-MS
- DAGAN : simply apply data augmentation (rotation) in a differentiable manner *(to disentangle the effect of augmentation and SSL)*
- DAGAN-MD: DAGAN has augmentation leakage issue, as augmentation is uniformly sampled. Hence, to guarantee convergence, keep multiple discriminator head for each augmentation. (SSL signal is fed as input index, not to-be-predicted target)

### Distribution Learning

![image](https://user-images.githubusercontent.com/46308407/147865984-2fa9854b-d992-42e7-8bb9-4f6dfbb86899.png)
![image](https://user-images.githubusercontent.com/46308407/147866021-47c8302d-0b29-4723-ac16-2195b868594b.png)

- GAN refers to unconditional BigGAN.
- All models are unconditional; no class label provided.
- To suppress augmentation leakage, up-weight the probability for identity transformation and dub DAGAN+. (similar to StyleGAN-ADA)

### Representation Learning

![image](https://user-images.githubusercontent.com/46308407/147866049-237e76e0-e6ab-458f-9fd4-24d7e05dca7f.png)

### Catastrophic Forgetting

![image](https://user-images.githubusercontent.com/46308407/147866061-e58fbc53-3e59-446c-b145-ea5d05363765.png)

- CIFAR-10 top-1 accuracy from representations of D.

### Data-Limited Regime

![image](https://user-images.githubusercontent.com/46308407/147866077-858b81f9-564a-47a4-9311-14d4f7b718b3.png)

- SSL helps data-limited distribution learning. (disentangled from the effect of augmentation to *some* extent.)

## Comments

- Interesting analogy of continual learning and GAN training.
- The idea is very simple; have the discriminator model SSL and adversarial objective jointly (2k-class)
- The evaluation, in my opinion, isn't particularly impressive, as they show only limited amount of samples.
- However, modeling complex joint data distribution such as ImageNet unconditionally is a demanding task, and encouraging discriminator to reason about general semantic is an important research direction.
